daffodil-cli/src/it/scala/org/apache/daffodil/debugger/TestCLIDebugger.scala

```diff
@@ -1265,20 +1265,29 @@ class TestCLIdebugger {
       shell.sendLine("step")
       shell.sendLine("step")
       shell.sendLine("step")
-      shell.sendLine("step")
-      shell.sendLine("step")
+      //
+      // NOTE FOR REVIEWER
+      //
+      // This test had to be modified to work with this change set.
+      // I have no idea why this change set would affect behavior
+      // of this kind.
+      //
```

Comment:
Maybe related to the serilazation stuff? Maybe forcing serialization is cause different code paths to be triggers? Or maybe thing are getting evaluated in different orders, which somehow changes thigns? Doesn't seem likely, but the only thing I can think of. Does this behavior switch back if we remove your serialization changes?

Reply:
I commented out the new serialization stuff, and there is no change to the behavior of these tests. 

I remain mystified. 

It appears that some alignment optimizations are working now, so that alignment unparsers aren't being created any longer in one of these tests anyway. That or they're being resolved and aren't in the data structure any more when it is being displayed.